### PR TITLE
Feat: add orm opentelemetry filter

### PR DESCRIPTION
**Pull request by Dokiys**
issue:#4944 
I've referred `opentracing/filter.go` to add this pr. But i'm not sure how to test this featrue, although there is an unit test.

### CODE REVIEW

File: client/orm/filter/opentelemetry/filter.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+package opentelemetry
+
+import (
+	"context"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	otelTrace "go.opentelemetry.io/otel/trace"
+	"strings"
+
+	"github.com/beego/beego/v2/client/orm"
```
Comment (flycash) [inline client/orm/filter/opentelemetry/filter.go:24]: please format these imports

File: client/orm/filter/opentelemetry/filter.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+// buildSpan add default span attributes and custom attributes with customSpanFunc
+func (builder *FilterChainBuilder) buildSpan(span otelTrace.Span, ctx context.Context, inv *orm.Invocation) {
+	span.SetAttributes(attribute.String("orm.method", inv.Method))
+	span.SetAttributes(attribute.String("orm.table", inv.GetTableName()))
+	span.SetAttributes(attribute.Bool("orm.insideTx", inv.InsideTx))
+	v, _ := ctx.Value(orm.TxNameKey).(string);
+	span.SetAttributes(attribute.String("orm.txName", v))
+	span.SetAttributes(attribute.String("span.kind", "client"))
+	span.SetAttributes(attribute.String("component", "beego"))
+
```
Comment (flycash) [inline client/orm/filter/opentelemetry/filter.go:69]: remove the unnecessary empty line

File: client/orm/filter/opentelemetry/filter.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+func NewFilterChainBuilder() *FilterChainBuilder {
+	return &FilterChainBuilder{}
+}
+
+func (builder *FilterChainBuilder) SetCustomSpanFunc(spanFunc CustomSpanFunc) *FilterChainBuilder {
+	builder.customSpanFunc = spanFunc
+	return builder
+}
```
Comment (flycash) [inline client/orm/filter/opentelemetry/filter.go:42]: remove this function and change to use option pattern. You can learn the Go option pattern via google

File: client/orm/filter/opentelemetry/filter.go (hunk 2 above)
Comment (Dokiys) [reply client/orm/filter/opentelemetry/filter.go:78]: 👌OK

**Comment by Dokiys** [comment]
@flycash Hello, I fix the test. But about golangci-lint error I have no idea. I'm not sure which file cause the error. I use gofumpt and gofmt to check both files there is no changes.

**Comment by flycash** [comment]
you can click the `Details` for more information. 
And do remember add this MR into change log

**Comment by flycash** [comment]
And you can find the golant-ci issues in `File changed`

**Comment by Dokiys** [comment]
> And you can find the golant-ci issues in `File changed`

@flycash Thanks. I've modified again.

**Comment by flycash** [comment]
image
Hi, can you help to fix this?

**Comment by Dokiys** [comment]
@flycash hi, I tried my best to fix this, but I still don't know why these code cause the error.  I just add as `client/orm/filter/opentracing/filter.go`.

File: client/orm/filter/opentelemetry/filter.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+		defer span.End()
+
+		res := next(spanCtx, inv)
+		builder.buildSpan(span, spanCtx, inv)
+		return res
+	}
+}
+
+// buildSpan add default span attributes and custom attributes with customSpanFunc
+func (builder *FilterChainBuilder) buildSpan(span otelTrace.Span, ctx context.Context, inv *orm.Invocation) {
```
Comment (flycash) [inline client/orm/filter/opentelemetry/filter.go:71]: please make the `ctx` as the first parameter

**Review by flycash — CHANGES_REQUESTED** [review]

**Pull request merged**